api/evidence/evidence.go

```diff
@@ -41,40 +42,37 @@ func (req *StoreEvidenceRequest) GetPayload() proto.Message {
 
 func (ev *Evidence) GetOntologyResource() ontology.IsResource {
 	var (
-		m        proto.Message
 		resource ontology.IsResource
-		err      error
 		ok       bool
 	)
-	// TODO: find a smarter way, because now we are unmarshalling the resource twice
-	// Try to extract the resource out of the evidence
-	m, err = ev.Resource.UnmarshalNew()
-	if err != nil {
+
+	if ev.Resource == nil || ev.Resource.Type == nil {
 		return nil
 	}
 
-	resource, ok = m.(ontology.IsResource)
+	// A little bit of dark Go magic
+	typ := reflect.ValueOf(ev.Resource.Type).Elem()
```

Comment:
I saw that this `.Elem()` function could panic. Therefore I took a closer look and added some tests. It's good due to the nil checks above.